File: app/src/main/java/io/github/getsmp/lorforandroid/ui/section/gallery/GalleryItem.java
/*
 * Copyright 2016 getsmp
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package io.github.getsmp.lorforandroid.ui.section.gallery;

public class GalleryItem {
    private final String image;
    private final String url;
    private final String title;
    private final String groupTitle;
    private final String postDate;
    private final String commentsCount;
    private final String tags;
    private final String author;

    public GalleryItem(String url, String title, String groupTitle, String postDate, String commentsCount, String tags, String author, String image) {
        this.url = url;
        this.title = title;
        this.groupTitle = groupTitle;
        this.postDate = postDate;
        this.commentsCount = commentsCount;
        this.tags = tags;
        this.author = author;
        this.image = image;
    }

    public String getUrl() {
        return url;
    }

    public String getTitle() {
        return title;
    }

    public String getPostDate() {
        return postDate;
    }

    public String getCommentsCount() {
        return commentsCount;
    }

    public String getTags() {
        return tags;
    }

    public String getAuthor() {
        return author;
    }

    public String getGroupTitle() {
        return groupTitle;
    }

    public String getImage() {
        return image;
    }
}

File: app/src/main/java/dev/trubitsyn/lorforandroid/ui/section/gallery/GalleryItem.kt
/*
 * Copyright (C) 2015-2021 Nikola Trubitsyn (getsmp)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

package dev.trubitsyn.lorforandroid.ui.section.gallery

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
data class GalleryItem(
        @PrimaryKey(autoGenerate = true)
        val id: Long = 0,
        val url: String,
        val title: String,
        val groupTitle: String?,
        val tags: String,
        val date: String,
        val author: String,
        val comments: Int,
        val imageUrl: String,
        val medium2xImageUrl: String,
        val mediumImageUrl: String
)
